### ISSUE TITLE
NamespaceIdAsNumber and InstanceIdAsNumber return null

### ISSUE THREAD

**Issue by ilkkapeltola** [andijakl/universal-beacon#1]
Running your sample code, both NamespaceIdAsNumber and InstanceIdAsNumber return null.
I'm running an iBKS beacon set to transmit Eddystone UID Frame, which gets detected correctly, but the NamespaceId and InstanceId are always null.

By the way thanks for working on this, a huge improvement on the WP beacon front!

**Comment by andijakl** [comment]
Can you maybe share the byte array contents of these IDs, without the AsNumber conversions?
I've been using the Kontakt.io Eddystone beacons to successfully test the conversion.

Especially the namespace ID is pretty straightforward, as it only reverses the array (big / little endian conversion) and converts to a Big Integer (as the namespace ID has 10 bytes according to the specification) - there's not much that could go wrong here.

**Comment by ilkkapeltola** [comment]
I'd love to!
Do you mean the ones available with just NamespaceId and InstanceId? They are null, too.

I've implemented an Eddystone solution on an Android device, and that has no issues, so I'm fairly certain that the beacons are at least to a degree working ;)

My original plan was (still is) to intercept the BluetoothLEAdvertisementReceivedEventArgs in my background task and build a Universal Beacon class based on the args. That way I would get all the benefits I need from your library. But now as I'm stuck, I'm trying to debug it.

How might I go about debugging the advertisement payloads?
I'm trying to dig into the Advertisements data, and I've got that far that I can catch a dataSection in my Advertisement.DataSections, which has a DataType == 0x16 (as per your comments in the library). But my skills run out trying to dig into the dataSection.Data.

**Comment by andijakl** [comment]
I can analyze what the beacons that you're using send - would be interesting to check that and to cross-reference that with the Eddystone specification! What I'd need for that would be the payload of the data array (from where you've found it already) - can you check that with the Visual Studio Debugger and post the raw data?

**Comment by ilkkapeltola** [comment]
if (dataSection.DataType == 0x16)
    Debug.WriteLine(BitConverter.ToString(dataSection.Data.ToArray()));

I get randomly one of the following three results:
AA-FE-00-D7-0F-87-BD-B0-92-70-45-55-8C-1D-AA-D0-89-8A-4B-B0-00-00
AA-FE-10-D7-02-67-6F-6F-2E-67-6C-2F-79-62-36-4D-67-74
AA-FE-20-00-0C-75-21-4B-00-00-00-80-00-00-04-B0

I'm not at all sure what to make of this.

**Comment by andijakl** [comment]
Thanks for the ID info! If the first byte after the AA-FE identifier is 00, it's the UID frame - so that is the one we're interested in for the IDs. I'll check out the exact contents of the frame as soon as possible to see if it corresponds to the standard and / or if there is an issue in the library!

**Comment by ilkkapeltola** [comment]
Thanks

**Comment by andijakl** [comment]
Found & fixed the issue, will update the NuGet package and the sample app in the store as well!

Problem was the validity check for the Eddystone UID frame. The length of your frame did not match the expected length. Interestingly, your frame is 100% according to specification, whereas the kontakt.io beacons I'm using are omitting the last two bytes (which are 0x00 0x00 and reserved for future use). I've now changed the class to be more flexible and to allow both variants! When creating a UID frame from code, the library will now add those two bytes to the UID frame as well.

Thanks for finding the issue, and sorry that it took a while until I could find the time to dig into it!
